src/View/ViewAbstract.php

```diff
@@ -1,281 +1,48 @@
 <?php
-declare(strict_types = 1);
-/**
- * This file is part of YAPEPBase.
- *
- * @copyright  2011 The YAPEP Project All rights reserved.
- * @license    http://www.opensource.org/licenses/bsd-license.php BSD License
- */
+declare(strict_types=1);
+
 namespace YapepBase\View;
 
 use YapepBase\Application;
-use YapepBase\Exception\Exception;
-use YapepBase\Storage\IStorage;
+use YapepBase\View\Data\ICanEscape;
 
 /**
  * ViewAbstract class what should be extended by every View class.
  */
-abstract class ViewAbstract
+abstract class ViewAbstract implements IRenderable
 {
-    /**
-     * The ViewDo instance used by the view
-     *
-     * @var \YapepBase\View\ViewDo
-     */
-    protected $viewDo;
-
-    /**
-     * Stores the content type
-     *
-     * @var string
-     */
-    protected $contentType;
-
-    /**
-     * Storage object which will be used for caching the rendered view object.
-     *
-     * @var \YapepBase\Storage\IStorage
-     */
-    private $storage;
-
-    /**
-     * The key which will be used for storing the rendered view object.
-     *
-     * @var string
-     */
-    private $storageKey;
-
-    /**
-     * Time to leave in seconds for storing the view object.
-     *
-     * @var int
-     */
-    private $storageTtl;
+    /** @var ICanEscape */
+    private $data;
```

Comment:
Not sure if this belongs on the abstract. Blocks should have already escaped values passed into them

Reply:
Yeah, but layouts and components are using this as well.